### PR TITLE
Optimize recent orders and trades fetching

### PR DESCRIPTION
**Pull request by antonyshchenko**
Order table in demo has around 70M records and fetching recent orders for market maker takes minutes affecting overall database performance quite badly.

### CODE REVIEW

File: backend/src/main/kotlin/xyz/funkybit/core/model/db/migrations/V98_AddUserGuidToOrderAndExecution.kt [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+import xyz.funkybit.core.db.Migration
+
+@Suppress("ClassName")
+class V98_AddUserGuidToOrderAndExecution : Migration() {
+    override fun run() {
+        transaction {
+            exec(
+                """
+                ALTER TABLE "order" ADD COLUMN user_guid CHARACTER VARYING(10485760);
+                UPDATE "order" SET user_guid = (
```
Comment (antonyshchenko) [inline backend/src/main/kotlin/xyz/funkybit/core/model/db/migrations/V98_AddUserGuidToOrderAndExecution.kt:13]: This takes around 30 minutes locally on demo db snapshot so I propose to reset the demo env.

File: backend/src/main/kotlin/xyz/funkybit/core/model/db/OrderExecution.kt [hunk 2]
```diff
@@ -118,46 +120,36 @@ class OrderExecutionEntity(guid: EntityID<ExecutionId>) : GUIDEntity<ExecutionId
...
 
         fun listForUser(userId: EntityID<UserId>, beforeTimestamp: Instant, limit: Int): List<OrderExecutionEntity> {
             return OrderExecutionTable
-                .join(OrderTable, JoinType.INNER, OrderTable.guid, OrderExecutionTable.orderGuid)
-                .innerJoin(WalletTable)
                 .join(TradeTable, JoinType.INNER, TradeTable.guid, OrderExecutionTable.tradeGuid)
                 .select(OrderExecutionTable.columns)
                 .where {
-                    WalletTable.userGuid.eq(userId) and OrderExecutionTable.timestamp.less(beforeTimestamp)
+                    OrderExecutionTable.userGuid.eq(userId) and OrderExecutionTable.timestamp.less(beforeTimestamp)
```
Comment (imykolenko) [inline backend/src/main/kotlin/xyz/funkybit/core/model/db/OrderExecution.kt:148]: it looks like composite index by `userId` and `timestamp` could be useful here

File: backend/src/main/kotlin/xyz/funkybit/core/model/db/OrderExecution.kt [hunk 3]
```diff
@@ -118,46 +120,36 @@ class OrderExecutionEntity(guid: EntityID<ExecutionId>) : GUIDEntity<ExecutionId
...
-                .innerJoin(WalletTable)
                 .select(TradeTable.responseSequence)
                 .withDistinct(true)
                 .where {
-                    WalletTable.userGuid.eq(userId)
+                    OrderExecutionTable.userGuid.eq(userId)
                 }
                 .orderBy(TradeTable.responseSequence, SortOrder.DESC)
                 .limit(maxSequencerResponses)
                 .mapNotNull { it[TradeTable.responseSequence] }
```
Comment (imykolenko) [inline backend/src/main/kotlin/xyz/funkybit/core/model/db/OrderExecution.kt:130]: This query filters and sorts by different tables. Coming to a result means either scaning backwards by response sequence and collect limit records with matching userID OR build a virtual list of all executions for a given user and then sort it by sequence number and then take limit. Both can take a while.

To solve this I'd propose to store `responseSequence` also on OrderExecution, then it becomes a query on one single table/index.

File: backend/src/main/kotlin/xyz/funkybit/core/model/db/migrations/V98_AddUserGuidToOrderAndExecution.kt (hunk 1 above)
Comment (imykolenko) [reply backend/src/main/kotlin/xyz/funkybit/core/model/db/migrations/V98_AddUserGuidToOrderAndExecution.kt:13]: demo reset is rest, should be quick now :)

**Review by imykolenko — APPROVED** [review]
looks great, nice work optimizing data access speed!

File: backend/src/main/kotlin/xyz/funkybit/core/model/db/OrderExecution.kt (hunk 2 above)
Comment (antonyshchenko) [reply backend/src/main/kotlin/xyz/funkybit/core/model/db/OrderExecution.kt:162]: added

File: backend/src/main/kotlin/xyz/funkybit/core/model/db/OrderExecution.kt (hunk 3 above)
Comment (antonyshchenko) [reply backend/src/main/kotlin/xyz/funkybit/core/model/db/OrderExecution.kt:144]: done

**Pull request merged**